phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/common/widgets/lists/ListWidgetEntry.java

```diff
@@ -105,7 +117,28 @@ public ListWidgetEntry(E item, Optional<URI> iconUri, URI defaultIconUri, String
     }
 
     public static ListWidgetEntry<ContainerDTO> create(ContainerDTO container) {
-        return new ListWidgetEntry<>(container, Optional.empty(), StaticMiniature.CONTAINER_MINIATURE,
+        Optional<URI> shortcutMiniature = Optional.empty();
+
+        final List<BufferedImage> miniatures = new ArrayList<>();
+        // do not use too many segments (cannot recognize the miniature if the segment is too small)
+        final int maxSegments = 4;
+        int currentSegment = 0;
+        for (ShortcutDTO shortcutDTO : container.getInstalledShortcuts()) {
```

Comment:
This begs for `Stream::takeWhile`...
Sadly `takeWhile` is coming with Java 9...